### PR TITLE
[feat] fused argmax + softmax-probability-of-argmax kernel

### PR DESCRIPTION
**Pull request by wenxuewuhd**
### Motivation

Confidence-driven decoders need, per row, two things: the argmax token, and the softmax probability of that same token — the score compared against an unmask or acceptance threshold. Diffusion LLMs do this on every denoise step.

Computed portably that is a `max` / `sub` / `exp` / `sum` / `gather` chain over `[rows, vocab]`. It reads the logits five to six times and materializes full-width temporaries, which at a 157k vocab and a few thousand rows both dominates the step and caps the batch size — one fp32 `[rows, vocab]` copy is 2.1 GiB at 3584 rows.

### Change

`sgl_kernel_npu/sample/argmax_softmax_prob.py` adds `argmax_softmax_prob_fused(logits) -> (argmax_ids, prob)`.

The kernel reads each logit once, carrying running `(max, argmax, sumexp)` in registers via the online-softmax recurrence, and returns `(argmax_id, 1/sumexp)`. The identity it exploits: the argmax token's logit *is* the running max, so its shifted exponent `exp(max - max)` is exactly 1 and its softmax probability is exactly the reciprocal of the denominator — the probability falls out of the same pass, with no second reduction and no `[rows, vocab]` write.

Accumulation is fp32 regardless of input dtype. Ties keep the lower index, matching `torch.argmax`, including across a chunk boundary.

An arbitrary row stride is accepted (only the last dim must be contiguous), so a vocab-truncated view can be passed without a copy.

### Tile sizing

Two bounds, both found by the tests rather than by inspection:

- **Dtype-derived width.** The vocab tile is sized against a 32 KiB budget (`_TILE_BYTES // logits.element_size()`), not a fixed element count. A 16384-element tile compiles for 2-byte dtypes but overflows the unified buffer for fp32 — `bishengir` reports the extra local buffer auto-multi-buffering needs. Measured on 910B3: fp32 at 16384 fails, at 8192 compiles; bf16 and fp16 compile at all three widths tested.
- **Clamped to the vocab.** `min(block_v, next_power_of_2(V))`. Without it a vocab narrower than the tile fails to compile once the single-iteration loop is unrolled.

### Accuracy

Measured on Ascend 910B3 at vocab 157184, bf16 logits, against an fp32 `argmax` + `softmax` + `gather` reference:

| Quantity | Result |
| --- | --- |
| argmax id | identical on every row |
| probability | max relative error 3.5e-7 |
| decisions at a 0.5 threshold | no flips |

### Speed

Against a chunked in-framework reduction over the same logits: **6.6x at 32 rows** (launch-bound) and **2.7x at 4096 rows**. Against the fp32 copy plus `argmax`/`softmax`/`gather` it replaces in a diffusion-LLM denoise step: 12.4 ms → 5.0 ms at 4096 rows.

### Tests

`tests/python/sgl_kernel_npu/test_argmax_softmax_prob.py`, following the existing kernel-test style in that directory (plain pytest functions against a torch golden):

- parametrized over `{bfloat16, float16, float32}` x `{(32, 4096), (128, 157184), (1024, 32000)}` — argmax bit-exact, probability within `1e-5`;
- a vocab below the tile width (the `next_power_of_2` clamp);
- a row stride wider than the vocab (the no-copy view case);
- an exact tie spanning a chunk boundary (lower index wins, as `torch.argmax` does).

```
$ PYTHONPATH=python/sgl_kernel_npu python -m pytest \
    tests/python/sgl_kernel_npu/test_argmax_softmax_prob.py -q
12 passed, 14 warnings in 17.07s
```

The dtype parametrization is what caught the fp32 tile overflow; the first two cases in the list are each a compile failure that reached the test before it reached a caller.

### CI

No CI change needed. No workflow under `.github/workflows/` runs `tests/python/sgl_kernel_npu/**` — the NPU workflows (`pr-test-deepep-npu.yml`, `a2-internode-test.yml`, `daily-build-test.yml`, `internode.yml`) only invoke `tests/python/deepep/*`, and their `paths` filters do not match this PR. `lint.yml` (pre-commit) is the only workflow that applies, and it passes.

### CODE REVIEW

**Review by ping1jing2 — APPROVED** [review]

**Comment by HEX1A0A** [comment]
approve
